Automattic/jetpack
# CRM: Automations - Adding Tag List data type, transformer, and Tag condition

## Proposed changes:

* This PR creates both a Tag and Tag List data type, as well as a transformer class to get the tag data from any object (data type).
  * There are some assumptions here, the main one being in terms of the structure of a tag list (including an ID and then an array of tags), which may need altering.
* This PR also adds a Tag condition and test, in order to better test the Tag List data type.
* What it doesn't have at this stage is a transformer test (we'll leave this for now)

### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/3309
https://github.com/Automattic/zero-bs-crm/issues/3310

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Currently the tag condition can be tested amongst all CRM tests - `jetpack docker phpunit-crm` (make sure to `jetpack build plugins/crm` or `composer dumpautoload` in the `crm/` folder first). The new test tests if a given tag matches a given condition, making use of the Tag and Tag List data types.
